src/Guzzle/MiddleWare/RequestLogger.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace Acquia\ContentHubClient\Guzzle\Middleware;
+
+use GuzzleHttp\Psr7\Request;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class RequestLogger.
+ */
+class RequestLogger
+{
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * RequestLogger constructor.
+     *
+     * @param \Psr\Log\LoggerInterface $logger
+     */
+    public function __construct(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    /**
+     * Logs any request.
+     *
+     * @param callable $handler
+     *
+     * @return \Closure
+     */
+    public function __invoke(callable $handler): \Closure
+    {
+        return function (Request $request, array $options) use ($handler) {
+
+            $promise = function (ResponseInterface $response) use ($request) {
+                try {
+                    $this->logResponse($request, $response);
+                } catch (\Exception $exception) {
+                    $message = sprintf('Failed to log request. Reason: %s', $exception->getMessage());
+                    $this->logger->critical($message);
+                }
+
+                return $response;
+            };
+
+            return $handler($request, $options)->then($promise);
+        };
+    }
+
+    /**
+     * Logs response.
+     *
+     * @param \GuzzleHttp\Psr7\Request $request
+     * @param \Psr\Http\Message\ResponseInterface $response
+     */
+    public function logResponse(Request $request, ResponseInterface $response): void {
+        if (!$this->isTrackable($response)) {
+            return;
+        }
+
+        $entry = implode('. ', $this->buildLogEntry($request, $response));
+
+        $responseStatusCode = $response->getStatusCode();
+        switch ($responseStatusCode) {
+            case $responseStatusCode >= 500:
+                $this->logger->error($entry);
+                break;
+            case $responseStatusCode >= 400:
+                $this->logger->warning($entry);
+                break;
```

Comment:
1- Use constants defined in Response class for 400 and 500
2- Switch statement is mainly for equality checks, not range check. Please use if/else instead.

Reply:
Refactored.